spyderlib/plugins/configdialog.py

```diff
@@ -903,77 +917,419 @@ def _save_lang(self):
+        self.schemes_combo.blockSignals(True)
+        names = self.get_option("names")
+        custom_names = self.get_option("custom_names")  # 'custom-n'
+
+        if custom_names:
+            choices = names + [None] + sorted(custom_names)
+            scheme_choices = list(zip(choices, [a for a in choices]))
+        else:
+            choices = sorted(names)
+            scheme_choices = list(zip(choices, [a.lower() for a in choices]))
+
+        combo = self.schemes_combo
+        combo.clear()
+
+        for item in scheme_choices:
+            name, key = item
+            if name is None and key is None:
+                continue
+            combo.addItem(name, key)
+
+        if custom_names:
+            combo.insertSeparator(len(names))
+
+        self.schemes_combo.blockSignals(False)
+        self.schemes_combo.setCurrentIndex(index)
+
+    def update_buttons(self):
+        """Updates the enable status of delete and reset buttons."""
+        current_scheme = self.current_scheme.lower()
+        names = [n.lower() for n in self.get_option("names")]
+        delete_visible = current_scheme not in names
+        self.delete_button.setEnabled(delete_visible)
+        self.reset_button.setEnabled(not delete_visible)
+
+    def update_preview(self, index=None, scheme_name=None):
+        """
+        Update the color scheme of the preview editor and adds text.
+
+        Note
+        ----
+        'index' is needed, because this is triggered by a signal that sends
+        the selected index.
+        """
+        text = ('"""A string"""\n\n'
+                '# A comment\n\n'
+                '# %% A cell\n\n'
+                'class Foo(object):\n'
+                '    def __init__(self):\n'
+                '        bar = 42\n'
+                '        print(bar)\n'
+                )
+        show_blanks = CONF.get('editor', 'blank_spaces')
+        if not scheme_name:
+            scheme_name = self.current_scheme
+        self.preview_editor.setup_editor(linenumbers=True,
+                                         markers=True,
+                                         tab_mode=False,
+                                         font=get_font('editor'),
+                                         show_blanks=show_blanks,
+                                         color_scheme=scheme_name)
+        self.preview_editor.set_text(text)
+        self.preview_editor.set_language('Python')
+
+    # Actions
+    # -------------------------------------------------------------------------
+    def create_new_scheme(self):
+        """Creates a new color scheme with a custom name."""
+        names = self.get_option('names')
+        custom_names = self.get_option('custom_names')
+        if not custom_names:
+            custom_names = []
+
+        # Get the available number this new color scheme
+        counter = len(custom_names) - 1
+        custom_index = [int(n.split('-')[-1]) for n in custom_names]

```

Comment:
Custom names for now is 

`custom-N` where N is calculated automatically, so for now there is no possibility of having a custom name....

This will change in the next PR related to this, as it required some extra changes that I considered too much for this pr alone